### PR TITLE
Poketch System Part 2

### PR DESCRIPTION
**Pull request by TrainerRiley**
This is as far as I think I can get in this file before documenting a couple other core Poketch files.

### CODE REVIEW

**Comment by lhearachel** [comment]
A couple things upfront:

1. Since this is a fresh review, could you squash out the merge commits with a rebase?
2. Is the separate author intentional?

**Comment by TrainerRiley** [comment]
Alright, cleaned up the history here. This history is still a little messed up (probably as a result of my previous failed rebase attempts yesterday), but it's only my changes now.

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:717]: I wonder if this matches:

```c
return poketchSys->appChanging == FALSE
    && ov25_0225450C(poketchSys) == 0
    && TouchScreen_TouchLocation(x, y)
    && PoketchSystem_InsideScreenBounds(*x, *y);
```

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:730]: Similar to above, I wonder if you can replace this with a conjunction-return.

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:740]: Unnecessary parens

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:774]: Use gender constant here for consistency

File: include/overlay025/poketch_system.h
Comment (TrainerRiley) [reply include/overlay025/poketch_system.h:25]: I commented it out because it's never actually used anywhere in the code (I think). But I'll uncomment it.

File: include/touch_screen.h
Comment (TrainerRiley) [reply include/touch_screen.h:43]: Yeah, naming these has been tough. I went with touch and tap becuase that terminology was already there, but I prefer tap and hold as well so I can change it.

File: src/overlay025/poketch_system.c
Comment (TrainerRiley) [reply src/overlay025/poketch_system.c:704]: Tried this and a bunch of variants, but it's not matching.

File: include/overlay025/poketch_button.h
Comment (cbt6) [inline include/overlay025/poketch_button.h:14]: > Slid touch off of button but did not release

What does this mean?

File: include/overlay025/poketch_system.h
Comment (cbt6) [inline include/overlay025/poketch_system.h:62]:
```suggestion
#define POKETCH_SCREEN_MIN_X   16
#define POKETCH_SCREEN_MIN_Y   16
#define POKETCH_SCREEN_MAX_X  207
#define POKETCH_SCREEN_MAX_Y  175
```

File: src/overlay025/poketch_system.c
Comment (cbt6) [inline src/overlay025/poketch_system.c:161]: What does a "field event" refer to here? Sounds different from that in `res/field/events`

File: src/overlay025/poketch_system.c
Comment (cbt6) [inline src/overlay025/poketch_system.c:586]:
```c
static void PoketchSystem_OnButtonEvent(u32 buttonID, u32 buttonEvent, u32 touchEvent, PoketchSystem *poketchSystem)
{
    if (!ov25_0225450C(poketchSystem)) {
```

File: include/overlay025/poketch_button.h
Comment (TrainerRiley) [reply include/overlay025/poketch_button.h:14]: This is a specific state for when someone starts pressing a button on the touch screen, but then moves the stylus away from the button without lifting it off the screen.

Comment (cbt6) [reply include/overlay025/poketch_button.h:14]: Maybe `BUTTON_MANAGER_STATE_DRAG` / `BUTTON_MANAGER_STATE_DRAGGING` then?

File: include/overlay025/poketch_system.h
Comment (lhearachel) [reply include/overlay025/poketch_system.h:25]: Feel free to leave it commented out, if it's currently unused. Or, alternatively, remove it.

File: src/overlay025/poketch_system.c
Comment (lhearachel) [reply src/overlay025/poketch_system.c:704]: Hm, you could take a look at the specific assembly for the function to see if there is a way to make it match. Unfortunate, if not.

**Comment by lhearachel** [comment]
We reached consensus on the naming scheme for functions moving forward. Please prefix your documented functions in this PR with appropriate verbage, per the naming scheme: `[<Module>_]<Imperative><Object>`

**Comment by TrainerRiley** [comment]
Okay, most comments addressed (besides the the ones trying to get the nested ifs to match with &&s instead, still working on that).

File: include/overlay025/poketch_system.h
Comment (lhearachel) [inline include/overlay025/poketch_system.h:79]: [polish] list the typical names of parameters to these functions

File: include/overlay025/poketch_system.h
Comment (lhearachel) [inline include/overlay025/poketch_system.h:132]: [polish] Thinking that these are more accessors than `Is` checks; maybe `PoketchSystem_GetDisplayHeldCoords` and `PoketchSystem_GetDisplayTappedCoords`?

File: include/touch_screen.h
Comment (lhearachel) [inline include/touch_screen.h:43]:
```suggestion
BOOL TouchScreen_GetHoldState(u32 *x, u32 *y);
BOOL TouchScreen_GetTapState(u32 *x, u32 *y);
```

My logic here is that `Get[Hold/Tap]State` makes it clear that this does more than just return the coordinates of the hold/tap, which is true; it will also return `TRUE` or `FALSE` depending on if the screen is actually held/tapped.

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:238]: [praise] Thanks for pulling this out of the function context!

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:326]: [quibble] Unnecessary parens

File: src/overlay025/poketch_system.c
Comment (lhearachel) [reply src/overlay025/poketch_system.c:519]: Did we determine if this matches?

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:555]: [polish] I would style this as:

```c
enum {
    POKETCH_BUTTON_TOP = 0,
    POKETCH_BUTTON_BOTTOM,
    POKETCH_BUTTON_SCREEN,
};

static const TouchScreenHitTable sMainPoketchButtons[] = {
    [POKETCH_BUTTON_TOP]    = {POKETCH_BUTTON_TOP_MINY,    POKETCH_BUTTON_TOP_MAXY,    POKETCH_BUTTON_TOP_MINX,    POKETCH_BUTTON_TOP_MAXX},
    [POKETCH_BUTTON_BOTTOM] = {POKETCH_BUTTON_BOTTOM_MINY, POKETCH_BUTTON_BOTTOM_MAXY, POKETCH_BUTTON_BOTTOM_MINX, POKETCH_BUTTON_BOTTOM_MAXX},
    [POKETCH_BUTTON_SCREEN] = {POKETCH_SCREEN_MINY,        POKETCH_SCREEN_MAXY,        POKETCH_SCREEN_MINX,        POKETCH_SCREEN_MAXX},
};
```

It is a bit more verbose, but leverages code syntax to both enforce a match and document what each element of this table corresponds to. You can also use the enum later for more semantic access to the table, if desired.

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:559]: [polish] Use a heap ID const for that `7` (`PoketchButtonManager_New` takes `heapID` for that parameter).

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:568]: [question] Does this match if you invert this conditional and use it as a sentinel?

```c
if (poketchSys->buttonManager == NULL) {
    return FALSE;
}

PoketchButtonManager_SetButtonTimer(poketchSys->buttonManager, 0, 0, 7);
poketchSys->unk_2C = 0xffffffff;
poketchSys->unk_30 = 0xffffffff;
return TRUE;
```

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:562]: [polish] If you go with my style suggestion above for `sMainPoketchButtons`, I would use the `enum` constant for that first `0` here.

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:604]:
```suggestion
        if (ov25_0225450C(poketchSys) && touchEvent == BUTTON_TOUCH_RELEASED) {
            Sound_PlayEffect(SEQ_SE_DP_BEEP);
        }
```

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:642]: [quibble] Unnecessary parens

File: src/overlay025/poketch_system.c
Comment (lhearachel) [inline src/overlay025/poketch_system.c:711]:
```suggestion
    if (poketchSys->appChanging == FALSE
        && ov25_0225450C(poketchSys) == 0)
        && TouchScreen_HoldLocation(x, y)) {
        return PoketchSystem_InsideScreenBounds(*x, *y);
    }
```

File: include/overlay025/poketch_system.h
Comment (lhearachel) [inline include/overlay025/poketch_system.h:135]: [polish] `PoketchSystem_GetPlayerMoved`

I'm not sold on the exact name of this struct-member; does it represent a player state that, e.g., updates the pedometer app's counter? Maybe `playerMoving` and `PoketchSystem_IsPlayerMoving`?

File: src/overlay025/poketch_system.c
Comment (TrainerRiley) [reply src/overlay025/poketch_system.c:559]: I'll fix up this one. Just checked and there's a bunch of other calls to this function in the poketch app files; I can save those for another PR.

File: include/overlay025/poketch_system.h
Comment (TrainerRiley) [reply include/overlay025/poketch_system.h:135]: After doing some digging, I agree that `playerMoving` and `IsPlayerMoving` are the right call. It looks a lot more like a state than a one-off event, and this is actually not used in the pedometer app, so other apps are probably checking the state (eg. dowsing machine).

File: src/overlay025/poketch_system.c
Comment (TrainerRiley) [reply src/overlay025/poketch_system.c:568]: This does not match.

Comment (lhearachel) [reply src/overlay025/poketch_system.c]: Unfortunate.

File: src/overlay025/poketch_system.c
Comment (TrainerRiley) [reply src/overlay025/poketch_system.c]: This does match, added in the most recent commit.

File: src/overlay025/poketch_system.c
Comment (TrainerRiley) [reply src/overlay025/poketch_system.c]: That doesn't match, but this does:

```if (!poketchSys->appChanging
    && !ov25_0225450C(poketchSys)
    && TouchScreen_GetTapState(x, y)) {
    return PoketchSystem_InsideScreenBounds(*x, *y);
}

return FALSE;````

File: src/overlay025/poketch_system.c
Comment (cbt6) [reply src/overlay025/poketch_system.c:578]: Bump

File: include/overlay025/poketch_system.h
Comment (cbt6) [reply include/overlay025/poketch_system.h]: Bump

File: include/overlay025/poketch_button.h
Comment (cbt6) [reply include/overlay025/poketch_button.h]: Remove the comment.

File: src/overlay025/poketch_system.c
Comment (TrainerRiley) [reply src/overlay025/poketch_system.c]: Renamed it to `PoketchSystem_SendEvent`.

Comment (cbt6) [reply src/overlay025/poketch_system.c]: Rename the enum to `PoketchEventID`.

File: src/overlay025/poketch_system.c
Comment (TrainerRiley) [reply src/overlay025/poketch_system.c:578]: I don't think we can do this one--or if we do, maybe in a separate PR. The function `PoketchButtonManager_New` requires this function signature, and that's called in almost every poketch app file so changing it would require changing function signatures in almost every app. This PR is already pretty big so I'm inclined to hold off for now.

**Comment by TrainerRiley** [comment]
Okay, I think we're good for real this time.

**Review by lhearachel — APPROVED** [review]
This looks mergeable to me. I'll give cbt a chance to look through again before merging.

File: include/overlay025/poketch_system.h
Comment (cbt6) [inline include/overlay025/poketch_system.h:73]:
```suggestion
enum PoketchEventID {
    POKETCH_EVENT_SLEEP = 0,
    POKETCH_VENT_PLAYER_MOVED,

    POKETCH_EVENT_SAVE = 4,
    POKETCH_EVENT_PEDOMETER,
};
```

**Pull request merged**
